### PR TITLE
#33: Update the front-end section to mention TypeScript (closes #33)

### LINKED ISSUE #33: Update the front-end section to mention TypeScript
## requirements
We officiali use TypeScript, but the front-end section mentions it only in a small note as something we sometimes rely on. We should update it to talk about TypeScript.

(Tutorial already updated here #32)

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ecamellini**
Closes #33

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.}

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: book/frontend/javascript/3.typing_js.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Taking for granted that the specification might never evolve to a better typed l
 
 ## Runtime type checks
 
-We are currently relying on the third option only, and we're doing so via [tcomb](https://github.com/gcanti/tcomb) and its ecosystem ([tcomb-react](https://github.com/gcanti/tcomb-react), [babel-plugin-tcomb](https://github.com/gcanti/babel-plugin-tcomb)).
+Before adopting TypeScript as our main language, we decided to leveragd [tcomb](https://github.com/gcanti/tcomb) and its ecosystem ([tcomb-react](https://github.com/gcanti/tcomb-react), [babel-plugin-tcomb](https://github.com/gcanti/babel-plugin-tcomb)) to perform runtime type checking.
```
Comment (gabro) [inline book/frontend/javascript/3.typing_js.md]: typo (lol)

`leveragd` -> `leverage`.

Occhio che non stiamo smettendo di usare tcomb, ma adesso complementa TS (vedi BRC)

Inoltre brucerei col fuoco `babel-plugin-tcomb` (lo stiamo usando ancora, @giogonzo?)

Comment (giogonzo) [reply book/frontend/javascript/3.typing_js.md]: nope, 🔪

**Review by gabro — APPROVED** [review]

File: book/frontend/guidelines/2.components.md [hunk 2]
```diff
@@ -8,12 +8,74 @@ Components are the building blocks of any react UI. They are not all the same th
...
+    onExpandToggle: () => {
+      transition({ postExpanded: !postExpanded });
+    }
+  })
+});
+```
+
+#### Plain Components
+
+Typically always (except for very rare exceptions) pure (i.e., should be `React.PureComponent`s or use the *deprecated* `@pure` decorator).
```
Comment (giogonzo) [inline book/frontend/guidelines/2.components.md]: I'd remove the `@pure` part altogether

File: book/frontend/guidelines/2.components.md [hunk 3]
```diff
@@ -8,12 +8,74 @@ Components are the building blocks of any react UI. They are not all the same th
...
+      <div className='post'>
+        {post}
+        <span onClick={onExpandToggle}>Click me!</span>
+      </div>
+    )
+  }
+}
+```
+
 #### Handlers (`RouteHandler`s)
```
Comment (giogonzo) [inline book/frontend/guidelines/2.components.md]: Not strictly related to this PR, but I'd remove this section completely 🔪

File: book/frontend/javascript/2.code_conventions_and_linting.md [hunk 4]
```diff
@@ -9,11 +9,11 @@ We try to standardize as much as possible the code we write with two main object
...
-There are some things an automated tool can't buy you. For everything else, there's [ESLint](https://github.com/eslint/eslint).
+There are some things an automated tool can't buy you. For everything else, there is [ESLint](https://github.com/eslint/eslint).
 
 [buildo/eslint-config](https://github.com/buildo/eslint-config) is a meticulously hand-crafted ESLint configuration that we use on all our projects. Except a few exceptions (you can always turn on/off rules for specific projects), it should exactly allow what we want and prevent much of the unwanted.
 
-It contains rules for JavaScript, rules specific to react, and some additional rules provided by our own plugins like [no-loops](https://github.com/buildo/eslint-plugin-no-loops) and [no-copy-paste](https://github.com/buildo/eslint-plugin-no-copy-paste-default-export).
+It contains rules for JavaScript, rules specific to react, and some additional rules provided by our own plugins like [no-loops](https://github.com/buildo/eslint-plugin-no-loops) and [no-copy-paste](https://github.com/buildo/eslint-plugin-no-copy-paste-default-export). [TSLint](https://palantir.github.io/tslint/) allows us to apply it also to TypeScript code.
```
Comment (giogonzo) [inline book/frontend/javascript/2.code_conventions_and_linting.md]: I'd add a note here, something like ", but we don't have a well maintained shared configuration yet. Stay tuned 😎 "

**Review by giogonzo — COMMENTED** [review]
Looks 💯 , left a few minor inline comments

**Review by giogonzo — APPROVED** [review]

**Pull request merged**